server/odc-service/src/main/java/com/oceanbase/odc/config/HookConfiguration.java

```diff
@@ -91,4 +108,20 @@ private void approvalFlowConfigUsageCheck(Long flowConfigId) {
         }
     }
 
+    private void projectReferenceCheck(Long userId, Long organizationId) {
+        Map<Long, Set<ResourceRoleName>> projectId2ResourceRoleNames =
+                resourceRoleService.getProjectId2ResourceRoleNames(userId);
+        if (projectId2ResourceRoleNames.size() == 0) {
+            return;
+        }
+        Map<Long, ProjectEntity> id2Project =
+                projectRepository.findAllByOrganizationId(organizationId).stream().filter(p -> p.getArchived() == false)
+                        .collect(Collectors.toMap(ProjectEntity::getId, p -> p));
+        throw new UnsupportedException(
```

Comment:
Error messages should ideally be internationalized, see `ErrorCodes.CannotOperateDueReference`

Reply:
okay, done~